example/poisson_gmg/poisson_package.hpp

```diff
@@ -0,0 +1,410 @@
+
+  auto pkg = md->GetMeshPointer()->packages.Get("poisson_package");
+  const auto alpha = pkg->Param<Real>("diagonal_alpha");
+
+  int nblocks = md->NumBlocks();
+  std::vector<bool> include_block(nblocks, true);
+
+  if (only_md_level) {
+    for (int b = 0; b < nblocks; ++b)
+      include_block[b] =
+          (md->grid.logical_level == md->GetBlockData(b)->GetBlockPointer()->loc.level());
+  }
+
+  auto desc = parthenon::MakePackDescriptor<in_t, out_t, div_t, rhs, D>(md.get());
+  auto pack = desc.GetPack(md.get(), include_block);
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "CaclulateFluxes", DevExecSpace(), 0, pack.GetNBlocks() - 1,
+      kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int b, const int k, const int j, const int i) {
+        const auto &coords = pack.GetCoordinates(b);
+        if ((i + j + k) % 2 == 1 && gs_type == GSType::red) return;
+        if ((i + j + k) % 2 == 0 && gs_type == GSType::black) return;
+        // Build the unigrid diagonal of the matrix
+        Real dx1 = coords.template Dxc<X1DIR>(k, j, i);
+        Real diag_elem =
+            -(pack(b, TE::F1, D(), k, j, i) + pack(b, TE::F1, D(), k, j, i + 1)) /
+                (dx1 * dx1) -
+            alpha;
+        if (ndim > 1) {
+          Real dx2 = coords.template Dxc<X2DIR>(k, j, i);
+          diag_elem -=
+              (pack(b, TE::F2, D(), k, j, i) + pack(b, TE::F2, D(), k, j + 1, i)) /
+              (dx2 * dx2);
+        }
+        if (ndim > 2) {
+          Real dx3 = coords.template Dxc<X3DIR>(k, j, i);
+          diag_elem -=
+              (pack(b, TE::F3, D(), k, j, i) + pack(b, TE::F3, D(), k + 1, j, i)) /
+              (dx3 * dx3);
+        }
+
+        // Get the off-diagonal contribution to Ax = (D + L + U)x = y
+        Real off_diag =
+            pack(b, te, div_t(), k, j, i) - diag_elem * pack(b, te, in_t(), k, j, i);
+
+        Real val = pack(b, te, rhs(), k, j, i) - off_diag;
+        pack(b, te, out_t(), k, j, i) =
+            weight * val / diag_elem + (1.0 - weight) * pack(b, te, in_t(), k, j, i);
+      });
+  return TaskStatus::complete;
+}
+
+template <class... vars>
+TaskStatus PrintChosenValues(std::shared_ptr<MeshData<Real>> &md,
+                             const std::string &label) {
+  using TE = parthenon::TopologicalElement;
+  auto pmb = md->GetBlockData(0)->GetBlockPointer();
+
+  auto desc = parthenon::MakePackDescriptor<vars...>(md.get());
+  auto pack = desc.GetPack(md.get());
+  std::array<std::string, sizeof...(vars)> names{vars::name()...};
+  printf("%s\n", label.c_str());
+  int col_num = 0;
+  for (auto &name : names) {
+    printf("var %i: %s\n", col_num, name.c_str());
+    col_num++;
+  }
+  const size_t scratch_size_in_bytes = 0;
+  const int scratch_level = 1;
+  const int ng = parthenon::Globals::nghost;
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "Print", DevExecSpace(), 0, pack.GetNBlocks() - 1, 0, 0, 0, 0,
+      KOKKOS_LAMBDA(const int b, int, int) {
```

Comment:
why two unused indices over a 1D range?

Reply:
I think the 1D loop wouldn't compile and I was writing this mostly for testing. I think the solution is just to remove this code (which I have done in #948.

Reply:
My guess if that it didn't compile because there was a mismatch between the explicit `DEFAULT_LOOP_PATTERN` and the implicit one for a 1D range.
Just calling it without should do the trick, otherwise it'd be a bug -- though not relevant as it's going to be removed as you say.